pkg/fqdn/dnsproxy/proxy_test.go

```diff
@@ -983,6 +984,15 @@ func (s *DNSProxyTestSuite) TestRestoredEndpoint(c *C) {
 	s.restoring = false
 }
 
+func (s *DNSProxyTestSuite) TestProxyRequestContext_IsTimeout(c *C) {
+	p := new(ProxyRequestContext)
+	p.Err = fmt.Errorf("sample err: %w", context.DeadlineExceeded)
+	c.Assert(p.IsTimeout(), Equals, true)
+
+	p.Err = fmt.Errorf("sample err: %s", context.DeadlineExceeded)
+	c.Assert(p.IsTimeout(), Equals, false)
```

Comment:
nitpicking: took me more time that I care to admit to see the difference between the two errors here (i.e. `%w` vs `%s`), maybe we could have a comment here like `// wrap the context.DeadlineExceeded error so that it can be found via errors.As` :sweat_smile: